src/main/java/nl/ordina/jobcrawler/service/VacancyService.java

```diff
@@ -32,8 +33,12 @@ public Vacancy add(Vacancy vacancy) {
         vacancy.setSkills(existingSkills); // add the skills to the vacancy
 
 
-        vacancy.checkURL(); //checking the url, if it is malformed it will throw a VacancyURLMalformedException
-        return vacancyRepository.saveAndFlush(vacancy); // save and send to the database
+        if (vacancy.checkURL()){    //checking the url, if it is malformed it will throw a VacancyURLMalformedException
+            return vacancyRepository.saveAndFlush(vacancy);
+        } else {
+            throw new VacancyURLMalformedException("Website could not be reached");
+        }
```

Comment:
This looks odd. What is happening here exactly? And what is supposed to happen?

Reply:
the name was changed to hasValidURL. So if a vacancy has a valid URL it saves it, else it throws an exception